### ISSUE TITLE
Style output via `tailwindcss-rails` gem appears off when compared to other build options

### ISSUE THREAD

**Issue by phillipspc** [tailwindlabs/tailwindcss#15426]
**What version of Tailwind CSS are you using?**

v3.4.16

**What build tool (or framework if it abstracts the build tool) are you using?**

the `tailwindcss-rails` gem

**What version of Node.js are you using?**

N/A

**What browser are you using?**

Issue seems browser agnostic, but I've tested Safari and Firefox

**What operating system are you using?**

macOS

**Reproduction URL**

Repo with minimal amount of changes to reproduce the issue here: https://github.com/phillipspc/railstailwind

**Describe your issue**

It looks like something is off in how styles are compiled via the `tailwindcss-rails` gem. When comparing the same HTML with the same utility classes against a Rails project building with Vite (minimal repo [here](https://github.com/phillipspc/railstailwindvite)), that styling looks correct there, but wrong in the gem version.

Here's a screenshot of a copy/pasted [TailwindUI component](https://tailwindui.com/components/application-ui/forms/input-groups#component-2607d970262ada86428f063c72b1e7bd):
![image](https://github.com/user-attachments/assets/4410fc8c-1dc5-4eda-b7f0-c396f4ae9d61)

I opened this [issue originally on the gem's GitHub page](https://github.com/rails/tailwindcss-rails/issues/443), however according to the maintainer:
> There's no magic in this gem, it's just running the upstream tailwindcss CLI against the generated tailwind config.

So it seems like whatever is breaking things in the gem is happening "upstream". This issue was [first reported](https://github.com/rails/tailwindcss-rails/discussions/441#discussion-7616317) on December 5th. 
I'm wondering if anyone from the Tailwind team would have a better idea, given that timing, of what might have changed that would introduce a downstream issue with the `tailwindcss-rails` gem. 

Any help is much appreciated, and if I can assist in providing any better reproduction/troubleshooting steps, I'm happy to do so!

**Comment by adamwathan** [comment]
Hey! I'm guessing it's this:

https://github.com/rails/tailwindcss-rails/blob/8a8dbd80b51c00ed54051bf6e125601c42c60b82/lib/install/tailwind.config.js#L18

The Rails gem is enabling the forms plugin by default which Tailwind UI doesn't expect and results in clashing styles.

**Comment by adamwathan** [comment]
Yeah personally I don't think any of those plugins should be in there and enabled by default, that's not the same experience you get when you initialize a Tailwind project outside of Rails and we don't actually use any of those plugins on our own projects. Should be opt-in for sure 👍

**Comment by phillipspc** [comment]
@adamwathan yup, that all checks out. I’ll pass that along and we can hopefully make that change to the gem in the near future. Thanks!
